server/src/main/java/com/google/sps/utils/StateSubcollectionBuilder.java

```diff
@@ -0,0 +1,138 @@
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.sps.utils.Ad;
+import com.google.sps.utils.Constants;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Description: Class creates collection by state. Each state collection contains
+ *              two subcollections: 'advertisers', which contains information about
+ *              each advertiser's total ad spend, and 'stateData', which contains
+ *              information about each state's total ad spend.
+ * Author: Kira Toal
+ * Date: July 20, 2020
+ */ 
+public class StateSubcollectionBuilder {
+
+  private static final String PATH_TO_SERVICE_ACCOUNT = "./serviceAccountKey.json"; 
+  private static final String DATABASE_URL = "https://step9-2020-capstone.firebaseio.com"; 
+  private static final String MAIN_COLLECTION = "ads";
+  private static final String WRITE_COLLECTION = "dev_states";
+  public static Firestore db; 
+
+  public static Firestore initializeApp() throws Exception {
+    // Set account and build options.
+    FileInputStream serviceAccount = new FileInputStream(PATH_TO_SERVICE_ACCOUNT);
+    FirebaseOptions options = new FirebaseOptions.Builder()
+        .setCredentials(GoogleCredentials.fromStream(serviceAccount))
+        .setDatabaseUrl(DATABASE_URL)
+        .build();
+    
+    // Initialize app.
+    if (FirebaseApp.getApps().isEmpty()) {
+      FirebaseApp.initializeApp(options);
+    }
+    db = FirestoreClient.getFirestore();
+    return db;
+  }
+
+  /**
+   * Maps advertisers to their total ad spend in a particular state
+   * @param documentsInState list of all document snapshots targeted
+   *                         at a particular state 
+   */
+  public static Map<String, Long> getAdvertiserToTotalSpend(
+      List<QueryDocumentSnapshot> documentsInState) {
+
+    Map<String, Long> advertiserToTotalSpend = new HashMap<>();
+    for (QueryDocumentSnapshot document: documentsInState) {
+      String advertiser = document.getString("advertiser");
+      long spendMax = document.getLong("spendMax");
+      if (advertiserToTotalSpend.containsKey(advertiser)) {
+        advertiserToTotalSpend.put(advertiser, 
+          advertiserToTotalSpend.get(advertiser) + spendMax);
+      } else {
+        advertiserToTotalSpend.put(advertiser, spendMax);
+      }
+    }
+    return advertiserToTotalSpend;
```

Comment:
Minor spacing nit: this is a pretty dense function/block of code (insert [chunky Putin](https://www.youtube.com/watch?v=nDanLjEnz7Q)). 

An easy way to improve readability would be to like  move line 69 to 68, ad whitespace between 'long' and 'if' and add whitespace between `return` and `}`.